### ISSUE TITLE
dunder method access to inner class __call__ ignores __get__ overloads

### ISSUE THREAD

**Issue by adhami3310** [microsoft/pyright#9574]
```py
from __future__ import annotations

from typing_extensions import Self, TypeVar, overload, reveal_type, TypeVarTuple

S = TypeVar("S")
R = TypeVarTuple("R")


class Function[*T]:
    def __init__(self, *prefilled_args):
        self.prefilled_args = prefilled_args

    def __call__(self, *args: *T) -> None:
        print(f"{self.prefilled_args + args=}")

    def partial_once[S, *R](self: Function[S, *R], arg: S) -> Function[*R]:
        return Function(*self.prefilled_args, arg)

    @overload
    def __get__(self, instance: None, owner) -> Self: ...

    @overload
    def __get__(self: Function[S, *R], instance: S, owner) -> Function[*R]: ...

    def __get__(self, instance, owner):
        if instance is None:
            return self
        return self.partial_once(instance)


class ArrayLike:
    __add__ = Function[Self, int]()


x = ArrayLike().__add__(1)
y = ArrayLike() + 1
```

Some of the code here is a bit much, but I wanted to illustrate some form of useful code that would use such functionality.

Both calls succeed just fine in runtime with the expected output:
```
self.prefilled_args + args=(<__main__.ArrayLike object at 0x7f2c575370e0>, 1)
```

Yet Pyright errors:
```
error: Operator "+" not supported for types "ArrayLike" and "Literal[1]" (reportOperatorIssue)
```
only for the second one.

It seems when using dunder methods, pyright is not evaluating the __get__ in the type checking.

MyPy fails both which I assume is more related to using TypeVarTuple, but Pyright seems to handle that better.

Note in the example above we have another typing error at `self.partial_once(instance)` but that one is correctly caught.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.391.
